pyramid/scripts/pserve.py

```diff
@@ -887,11 +889,28 @@ def check_reload(self):
                 continue
             if filename.endswith('.pyc') and os.path.exists(filename[:-1]):
                 mtime = max(os.stat(filename[:-1]).st_mtime, mtime)
+                pyc = True
+            else:
+                pyc = False
             if filename not in self.module_mtimes:
                 self.module_mtimes[filename] = mtime
             elif self.module_mtimes[filename] < mtime:
-                print("%s changed; reloading..." % filename)
-                return False
+                if pyc:
+                    filename = filename[:-1]
+                # check if a file has syntax errors. If so, track it until it's fixed
+                for filename_promise in self.syntax_error_files + [filename]:
+                    try:
+                        py_compile.compile(filename_promise, doraise=True)
+                    except py_compile.PyCompileError:
+                        print("%s has a SyntaxError; NOT reloading." % filename_promise)
```

Comment:
> Why is logging wrong? Logging is user-configurable and allows you to agnostically write to the user's preferred output. As far as informational versus erroneous conditions, that why log levels exist -- you can use info() and error().

Sorry but when I say "wrong" I mean "this isn't the current convention in Pyramid". This is how all of the `p*` scripts in pyramid currently work. This issue (let alone line note) are not the place to discuss changing that. Getting this pr accepted is about conforming to the current standard. If you want to initiate/discuss a refactor of all the scripts you can open a separate issue/PR about that.

As far as the verbosity goes it would help if you interspersed in your examples some of the other messages output by pserve such as `%s changed, reloading ...` and others to get a feel for what you're talking about in your last example. I agree with most of your points however. I haven't tried to run the code yet, and your output isn't from real code so it's a little hard to envision your proposed change. We could easily be talking past each other or agreeing.